File: Top 50 Coding Interview Questions/17-Write a code to find non-repeating elements in an array/find_non_repeating_elements.java
import java.util.Arrays;

public class find_non_repeating_elements {
   
    public static void countFreq(int arr[], int n){
        boolean visited[] = new boolean[n];
        Arrays.fill(visited, false);

        //traverse through array elements and 
        //count frequencies
        for( int i = 0; i < n; i++){
            //skip this element if already processed
            if(visited[i] == true)
              continue;

              //count frequency
        int count = 1;
        for(int j = i+ 1; j < n; j++){
            if(arr[i] == arr[j]){
                visited[j] = true;
                count++;
            }
        }
        if(count == 1)
           System.out.println(arr[i]);
        }
    }
    //Driver code
    public static void main(String[] args) {
       int arr[] = new int[] {10, 20, 30, 40, 20, 50, 10};
       int n = arr.length;
       countFreq(arr, n );
}
}

File: Top 50 Coding Interview Questions/17-Write a code to find non-repeating elements in an array/findnon-repeatingelements.py
#program to count unique elements
def count(arr, n):

    #Mark all arrays elements as not visited
    visited = [False for i in range(n)]

    #Traverse through array elements
    #and count frequencies
    for i in range(n):

        #skip this elements if already
        #processed
        if(visited[i] == True):
            continue

        #count frequency
        count = 1
        for j in range(i + 1, n ,1):
            if(arr[i] == arr[j]):
                visited[j] = True
                count +=1
        if count == 1:
            print(arr[i]);

#Driver code
arr = [10, 20, 40 ,30 , 10, 20 ,50 , 10]
n = len(arr)
count(arr, n)